code/hud/hudartillery.cpp

```diff
@@ -219,6 +252,9 @@ void ssm_create(object *target, vec3d *start, size_t ssm_index, ssm_firing_info
 	// Init the ssm data
 
 	count = Ssm_info[ssm_index].count;
+	if (Ssm_info[ssm_index].max_count != -1) {
+		count += rand32() % (Ssm_info[ssm_index].max_count - count + 1);
```

Comment:
Can you add a source code comment here explaining the logic behind the formula?
It would make understanding the code later much easier.